script.js

```diff
@@ -0,0 +1,96 @@
+
+filterTabs.forEach((filterTab) => {
+  filterTab.addEventListener("click", (event) => onFilterClick(filterTab));
+});
+
+const onFilterClick = (filterTab) => {
+  const oldSelectedTab = document.querySelector(".tab-selected");
+  oldSelectedTab.classList.remove("tab-selected");
+  filterTab.classList.add("tab-selected");
+};
+
+const scrollCarousel = (direction) => {
+  let width =
+    Math.floor(document.querySelector(".carousel-item").scrollWidth) + 32;
+  width = direction == "left" ? -width : width;
+
+  carousel.scrollBy({
+    left: width,
+    behavior: "smooth",
+  });
+};
+
+const carouselObserver = new IntersectionObserver(
+  ([entry]) => {
+    if (entry.isIntersecting) {
+      header.classList.remove("hide-header");
+      filterSection.classList.remove("filter-sticky");
+    }
+  },
+  {
+    root: null,
+    threshold: [0, 1],
+  }
+);
+
+const observer = new window.IntersectionObserver(
+  ([entry]) => {
+    if (entry.boundingClientRect.top <= 0) {
+      header.classList.add("hide-header");
+      filterSection.classList.add("filter-sticky");
+    }
+  },
+  {
+    root: null,
+    threshold: [0, 1],
+  }
+);
+
+observer.observe(filterSection);
+carouselObserver.observe(carouselContainer);
+
+rightButton.addEventListener("click", () => scrollCarousel("right"));
+leftButton.addEventListener("click", () => scrollCarousel("left"));
+
+let data = [
+  "https://res.cloudinary.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_264,h_288,c_fill/gfkicyyqxr0aegwrtyfh",
+  "https://res.cloudinary.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_264,h_288,c_fill/crumbigyyv72dwoepcxx",
+  "https://res.cloudinary.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_264,h_288,c_fill/soafwfvm6b7narqfemmu",
+  "https://res.cloudinary.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_264,h_288,c_fill/ghswgwzuh0lo8k4dn51m",
+  "https://res.cloudinary.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_264,h_288,c_fill/j2vhfamlnsk70zk1qhxb",
+  "https://res.cloudinary.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_264,h_288,c_fill/yiz4d7ztgtinx2xygbr5",
+  "https://res.cloudinary.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_264,h_288,c_fill/vts5euhkwodjpaq45nnm",
+  "https://res.cloudinary.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_264,h_288,c_fill/ghswgwzuh0lo8k4dn51m",
+  "https://res.cloudinary.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_264,h_288,c_fill/qfc7iuzrsckx4hujzujp",
+];
+
+data.forEach((src) => {
+  const card = document.createElement("div");
+  card.classList.add("card");
+  const element = `
+  <div class="content">
+    <img class="card-image"
+        src=${src} />
+    <div class="food-description">
+        <h3 class="food-name">Burger Point</h3>
+        <p class="food-cuisine">Healthy Food, Indian, Bakery</p>
+    </div>
+  </div>
+  <div class="hover-content">
+    <span class="hover-text">
+        I'm hovered!
+    </span>
+</div>
+  
+  `;
+  card.innerHTML = element;
```

Comment:
What's the difference between assigning the html string to the `innerHtml` property and using `createElement` with `appendChild`?

Reply:
`innerHTML` can be used to get or set the value of HTML element. To set `innerHTML`, we need to pass a string.The whole content of DOM Element will be replace with parsed value of string  . `appendChild` on the other hand take DOMNode as input and append to last in DOM element.It doesn't take string as input.

Reply:
What's the performance impact? Which is better when?

Reply:
`innerHTML += html-string` will first parse the complete element into string and parse it back into DOM nodes. This may affect the performance. `appendChild` doesn't affect other nodes in element. `appendChild` can be used when we don't want to rebuild the whole tree. `innerHTML` is appropriate when we want to replace the whole tree.